[SPARK-8429] [EC2] Add ability to set additional tags

armisael opened the pull request:
Add the `--additional-tags` parameter that allows to set additional tags to all the created instances (masters and slaves).

The user can specify multiple tags by separating them with a comma (`,`), while each tag name and value should be separated by a colon (`:`); for example, `Task:MySparkProject,Env:production` would add two tags, `Task` and `Env`, with the given values.

armisael committed:
Add ability to set additional tags

Add the `--additional-tags` parameter that allows to set additional tags to all the created instances (masters and slaves).

The user can specify multiple tags by separating them with a comma (`,`), while each tag name and value should be separated by a colon (`:`); for example, `Task:MySparkProject,Env:production` would add two tags, `Task` and `Env`, with the given values.

AmplabJenkins commented:
Can one of the admins verify this patch?

srowen commented:
Please read https://cwiki.apache.org/confluence/display/SPARK/Contributing+to+Spark

andrewor14 commented:
Hi @armisael, you need to file a JIRA here: https://issues.apache.org/jira/browse/SPARK. Once you have done that change the title of this PR to link against that JIRA? e.g.

```
[SPARK-XXXX] [EC2] Add ability to set additional tags
```

armisael commented:
Thank you @andrewor14 and @srowen and sorry if I missed that. Hope it's fine now.

srowen commented:
Sounds and looks reasonable; @nchammas ? @shivaram ?

nchammas commented:
See [SPARK-7900](https://issues.apache.org/jira/browse/SPARK-7900).

We have an issue with the tagging calls as they are sometimes getting hit by AWS slowness in propagating metadata about the instances.

I would favor as part of this PR bumping up the wait right before to 15 seconds or more.

armisael commented:
Increasing the sleep time would probably work, yes; interestingly I've been using this script for a while now and I've never faced this issue; anyway, I guess it may happen sometimes.

Let me also change the code slightly to set all the tags in a single call, to speed things up.

armisael committed:
Address SPARK-7900 by inceasing sleep time

We add tags (ex: Name) to instances right after creating them; we have to wait some time for their metadata to propagate, as sometimes the `add_tag` requests fail with a `InvalidInstanceID.NotFound`. With this commit we increase that sleep time as suggested by @nchammas: not a permanent solution, but it should help.

armisael committed:
Set multiple tags in a single request

Use `instance.add_tags` instead of `instance.add_tags` to set multiple tags at once, instead of firing multiple requests, one per tag.

andrewor14 commented:
ok to test

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #35165 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/35165/consoleFull) for   PR 6857 at commit [`8e614f1`](https://github.com/apache/spark/commit/8e614f1127e63b0214cdf127853d8a3e14df03a8).

SparkQA commented:
[Test build #35165 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/35165/console) for   PR 6857 at commit [`8e614f1`](https://github.com/apache/spark/commit/8e614f1127e63b0214cdf127853d8a3e14df03a8).
 * This patch **fails Python style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

armisael committed:
python style (pep8)

clean some leftovers that break the python style and make the build to fail

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #35180 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/35180/consoleFull) for   PR 6857 at commit [`c5ac92c`](https://github.com/apache/spark/commit/c5ac92c0ebb20af97f3623212c0158d94fcad67f).

SparkQA commented:
**[Test build #35180 timed out](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/35180/console)**     for PR 6857 at commit [`c5ac92c`](https://github.com/apache/spark/commit/c5ac92c0ebb20af97f3623212c0158d94fcad67f)     after a configured wait of `175m`.

AmplabJenkins commented:
Merged build finished. Test FAILed.

armisael commented:
Can anybody shed some light on why the build failed? I can't really understand what happened, it seems unrelated to the changes that this PR brings...

srowen commented:
Not related to this change, almost certainly

srowen commented:
jenkins, retest this please

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #35442 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/35442/consoleFull) for   PR 6857 at commit [`c5ac92c`](https://github.com/apache/spark/commit/c5ac92c0ebb20af97f3623212c0158d94fcad67f).

SparkQA commented:
[Test build #35442 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/35442/console) for   PR 6857 at commit [`c5ac92c`](https://github.com/apache/spark/commit/c5ac92c0ebb20af97f3623212c0158d94fcad67f).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

armisael commented:
Build issues again?

    Publisher 'Publish JUnit test result report' failed: No test report files were found. Configuration error?

shivaram commented:
Hmm I think the Junit publish error is a known bug as we have tried to decrease the number of tests we run per PR. (See http://apache-spark-developers-list.1001551.n3.nabble.com/jenkins-ERROR-Publisher-Publish-JUnit-test-result-report-failed-No-test-report-files-were-found-Conf-td12823.html)

cc @JoshRosen

JoshRosen commented:
@shivaram Yeah, ignore that error for now. I'll fix this soon with either a Jenkins conf. change or by adding a dummy test file so that the check never fails.  In the meantime, treat whatever SparkQA says as the source-of-truth for the tests and don't revert my test patch.

shivaram commented:
Thanks @JoshRosen -- @armisael @srowen so in this case the tests are fine as SparkQA commented they passed in https://github.com/apache/spark/pull/6857#issuecomment-114055404

shivaram commented:
@nchammas Any further comments ? Or is this good to go ?

nchammas commented:
@shivaram - I'll test it out now.

nchammas commented:
LGTM

shivaram commented:
Thanks @nchammas -- Merging this.
